File: 1768.go
package main

import "fmt"

func mergeAlternately(word1 string, word2 string) string {
	lenL := len(word1)
	lenR := len(word2)

	ret := make([]byte, 0, lenL+lenR)

	indexL := 0
	indexR := 0

	for indexL < lenL || indexR < lenR {
		if indexL < lenL {
			ret = append(ret, word1[indexL])
			indexL++
		}
		if indexR < lenR {
			ret = append(ret, word2[indexR])
			indexR++
		}
	}

	return string(ret)
}

func main() {
	word1 := "abc"
	word2 := "pqr"

	fmt.Println(mergeAlternately(word1, word2))
}

File: 1768.cpp
#include "shdstd-cpp.h"

class Solution
{
	public:
		std::string mergeAlternately(std::string word1, std::string word2)
		{
			std::string ret = "";

			int l, r;
			l = r = 0;

			while (l < word1.length() && r < word2.length()) {
				ret += word1[l++];
				ret += word2[r++];
			}

			while (l < word1.length()) {
				ret += word1[l++];
			}

			while (r < word2.length()) {
				ret += word2[r++];
			}

			return ret;
		}
};
